test_conformance/images/kernel_read_write/test_cl_ext_image_from_buffer.cpp

```diff
@@ -79,77 +79,69 @@ int image2d_from_buffer_positive(cl_device_id device, cl_context context,
-                                &base_address_alignment_2d, nullptr);
-            test_error(err, "Error clGetDeviceInfo");
+        cl_uint base_address_alignment_2d = 0;
+        err = clGetDeviceInfo(device, CL_DEVICE_IMAGE_BASE_ADDRESS_ALIGNMENT,
+                              sizeof(base_address_alignment_2d),
+                              &base_address_alignment_2d, nullptr);
+        test_error(err, "Error clGetDeviceInfo");
 
-            for (auto format : formatList)
-            {
-                cl_image_desc image_desc = { 0 };
-                image_desc_init(&image_desc, imageType);
+        for (auto format : formatList)
+        {
+            cl_image_desc image_desc = { 0 };
+            image_desc_init(&image_desc, imageType);
 
-                cl_mem_flags flag = (flagType == CL_MEM_KERNEL_READ_AND_WRITE)
-                    ? CL_MEM_READ_WRITE
-                    : flagType;
+            cl_mem_flags flag = (flagType == CL_MEM_KERNEL_READ_AND_WRITE)
+                ? CL_MEM_READ_WRITE
+                : flagType;
 
-                size_t row_pitch_alignment = 0;
-                size_t base_address_alignment = 0;
+            size_t row_pitch_alignment = 0;
+            size_t base_address_alignment = 0;
 
-                int get_error = get_image_requirement_alignment(
-                    device, context, flag, &format, &image_desc,
-                    &row_pitch_alignment, nullptr, &base_address_alignment);
-                if (TEST_PASS != get_error)
-                {
-                    return get_error;
-                }
+            int get_error = get_image_requirement_alignment(
+                device, context, flag, &format, &image_desc,
+                &row_pitch_alignment, nullptr, &base_address_alignment);
+            if (TEST_PASS != get_error)
+            {
+                return get_error;
+            }
 
-                const size_t element_size =
-                    get_format_size(context, &format, imageType, flag);
+            const size_t element_size =
+                get_format_size(context, &format, imageType, flag);
 
-                /*  Alignements in pixels vs bytes */
-                if (base_address_alignment
-                    > base_address_alignment_2d * element_size)
-                {
-                    test_fail("Unexpected base_address_alignment\n");
-                }
+            /*  Alignments in pixels vs bytes */
+            if (base_address_alignment
+                < base_address_alignment_2d * element_size)
+            {
+                test_fail("Unexpected base_address_alignment\n");
+            }
 
-                if (row_pitch_alignment > row_pitch_alignment_2d * element_size)
-                {
-                    test_fail("Unexpected row_pitch_alignment\n");
-                }
+            if (row_pitch_alignment < row_pitch_alignment_2d * element_size)
+            {
+                test_fail("Unexpected row_pitch_alignment\n");
```

Comment:
This check appears to be inverted as well, for the same reason.

```suggestion
            if (row_pitch_alignment > row_pitch_alignment_2d * element_size)
            {
                test_fail("Unexpected row_pitch_alignment\n");
```

Reply:
same question as above?